src/matlab2tikz.m

```diff
@@ -998,7 +998,9 @@ function addCustomCode(fid, before, code, after)
     backgroundColor = get(handle, 'Color');
     if ~isNone(backgroundColor)
         [m2t, col] = getColor(m2t, handle, backgroundColor, 'patch');
-        if ~strcmp(col, 'white')
+        % Fill background if not just white or if multiple axes and current
+        % is visible
+        if ~strcmp(col, 'white') || (numel(m2t.relevantAxesHandles) > 1 && isVisible(handle))
```

Comment:
While this is a good start, I wonder what happens with regular `subplot`s: wouldn't this add backgrounds there as well?
Another edge case, `plotyy` should work, since then one of the backgrounds should be `none`.

If `subplots` don't work as expected (i.e. only put the background is something non-default is set), we could check if the current axes overlaps with another relevant axes to exclude subplots from this kind of behavior.

However, I think very people won't notice the difference, since in 99% of the cases the paper background is also white.